### ISSUE TITLE
[CMake] not working with the last abseil-cpp release (20200923)

### ISSUE THREAD

**Issue by Mizux** [grpc/grpc#24438]
Currently grpc cmake-config or pkg-config file won't work with the last version of abseil since you depend on target `dynamic_annotation` which has been remove in `20200923`
https://github.com/grpc/grpc/blob/4645da201ae2c7d0b15fe56d86b41354fa4af0ca/CMakeLists.txt#L123

>  **Breaking Changes:**
Abseil no longer contains a `dynamic_annotations` library. Users
using a supported build system (Bazel or CMake) are unaffected by
this, but users manually specifying link libraries may get an error
about a missing linker input.

ref: https://github.com/abseil/abseil-cpp/releases/tag/20200923.1

related to: https://github.com/Homebrew/homebrew-core/issues/62589

ps: Are you in the abseil federation (if it's still exist)
pps: would you want a CL ?

**Comment by Mizux** [comment]
Concerning bazel build, it's a pain to figure out that you still use 20200225 since you don't even have a comment specifying which version you are using...
https://github.com/grpc/grpc/blob/4645da201ae2c7d0b15fe56d86b41354fa4af0ca/bazel/grpc_deps.bzl#L239-L248

Please take a look of what we have done for Google OR-Tools:
```bazel
load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")

git_repository(
    name = "com_google_absl",
    commit = "b56cbdd", # release 20200923
    remote = "https://github.com/abseil/abseil-cpp.git",
)
```
note: 20200923.1 is out, the commit shat1 is from github
see: https://github.com/abseil/abseil-cpp/releases (top left below the release tag)

**Comment by Mizux** [comment]
to remove also here:
https://github.com/grpc/grpc/blob/4645da201ae2c7d0b15fe56d86b41354fa4af0ca/CMakeLists.txt#L15909-L15916

end of line 15914...

**Comment by yulin-liang** [comment]
Looks like it's related to abseil-cpp. @veblush, can you help take a look?

**Comment by veblush** [comment]
Mizux, thank you for the bug. This bug is a known issue and we will address by

- Upgrading Abseil to LTS 20200923.2 #24270
- Remove our-own pkg-target installation once its counterpart in Abseil will be available in the next Abseil LTS version.

**Comment by Firefly35** [comment]
Any progress on the actions mentioned on october the 22 ?

**Comment by veblush** [comment]
This is resolved by #24270 so you can use it with the latest abseil-cpp release.
